x/rollapp/transfergenesis/ibc_module.go

```diff
@@ -107,39 +109,32 @@ func (w IBCModule) OnRecvPacket(
 	}
 
 	if !transfer.IsRollapp() {
-		l.Debug("Transfer is not from a rollapp.")
 		return w.IBCModule.OnRecvPacket(ctx, packet, relayer)
 	}
 
+	ra := transfer.Rollapp
+
 	memo, err := getMemo(transfer.GetMemo())
 	if errorsmod.IsOf(err, gerr.ErrNotFound) {
-		// If someone tries to send a transfer without the memo before the bridge is open, they will
-		// be blocked at the transfersenabled middleware
+		// The first regular transfer marks the full opening of the bridge, more genesis transfers will not be allowed.
+		w.rollappKeeper.EnableTransfers(ctx, ra.RollappId)
```

Comment:
it doesn't make any difference though

Reply:
wdym? this code will be called for every rollapp packet
```
func (k Keeper) EnableTransfers(ctx sdk.Context, rollappID string) {
	ra := k.MustGetRollapp(ctx, rollappID)
	ra.GenesisState.TransfersEnabled = true
	k.SetRollapp(ctx, ra)
	ctx.EventManager().EmitEvent(sdk.NewEvent(types.EventTypeTransferGenesisTransfersEnabled,
		sdk.NewAttribute(types.AttributeKeyRollappId, rollappID),
	))
}
```

Reply:
ah yeah I forgot it serialises the whole object
good catch thanks
probably we should not us huge objects as state values generally, where possible, in future

If this has just been writing a bool to db it would be totally fine